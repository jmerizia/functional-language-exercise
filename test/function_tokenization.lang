print a b => (
    if (equals s nil) then (nil)
    else (
        (print_int s:1);
        (print s:2)
    )
)

range n => (
    if (equals n 0) then [0, nil]
    else [n, (range (sub n 1))]
)

(print (range 5))
