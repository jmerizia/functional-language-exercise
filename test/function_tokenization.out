print function_declaration_name
s function_declaration_parameter
=> function_declaration_arrow
string123 function_declaration_expression
